---
title: Ryegrass, a potential source of gluten-like proteins
subtitle: 
short_title: 

author:  Sophia Escobar-Correas
affiliation: CSIRO Agriculture & Food # Or group/team
photo: Data/sophia.jpg

output: DSreport::project_summary
---
```{r setup, include=FALSE}
knitr::opts_chunk$set(
  results = 'asis',
  echo = FALSE,
  warning = FALSE,
  message = FALSE,
  fig.align = 'center'
)

#Load libraries
library(tidyverse)
library(gapminder)
library(kableExtra)
```


# Introduction
Hello! my name is Sophia, I am a molecular biologist working in proteomics, currently a Postdoctoral fellow.  Before Data School, I coded on Macro (Excel). I used to spend a lot of time cleaning and tidying protein data, I always felt like I could do it faster if I had programming skills. These weeks learning R have changed my daily work. The possible things that we could do have opened my mind to a new perspective of my research.

# My Project
I am working in the study of ryegrass, a potential source for gluten peptides contamination. **Gluten** refers to a class of storage proteins found in cereal grains, including wheat, rye, barley, and oats. Consumption of these gluten proteins leads to an autoimmune response in the case of coeliac disease. Previous studies have identified gluten-like proteins in ryegrass. Since this is a common weed found in grain fields, there is a possibility of cross-contamination. First of all, I need to characterize the gluten proteins and peptides with ryegrass origen. To do so I have performed Data-dependent mass spectrometric analysis. The results of this study provided identification of **3162 proteins**, and **8231 peptides**. Now what I need to do is found how many of them are gluten-like.  


## Preliminary results
To work with Protein Databases, that are usually in the format **.fasta**. I have used the package `Biostrings`. I will analyse aminoacid composition on the proteins of my database, since gluten proteins have a high composition of the aminoacids Glutamine (Q). I will search all proteins that have over 20% Glutamine and 10% Proline.
For tyding the data,`Tidyverse` (my new best friend) and `plyr`. 


**Tables**
```{r mytable, out.width='100%', echo = F}
my_database <- read.csv('Data/my_db.csv') 
knitr::kable(head(my_database, n = 5), format = "html", caption = "Protein Database") %>% 
  kable_styling("striped", full_width = F) %>%
  column_spec(4, width_max = "30em", extra_css="overflow-y: auto") %>% 
  column_spec(2:3, bold=T, color= "blue") %>% 
  footnote(general= "5 examples of proteins found in the database. The column Sequence indicates the amino acids (letter code) that make up each protein.",
          footnote_as_chunk = T)
```

**What we will do now is to look for the amino acids Q and P.**
```{r, echo=TRUE, message=FALSE}
library(tidyverse)
library(Biostrings)
my_database_comp <- mutate(my_database, totalAA= str_count(Sequence, '\\w'), Qcomp= str_count(Sequence,'Q')) %>% 
  mutate(Q100 =  Qcomp *100 / totalAA) %>% 
  mutate(Pcomp= str_count(Sequence,'P')) %>% 
  mutate(P100= Pcomp *100/totalAA)

my_database_comp[,'Q100']=round(my_database_comp[,'Q100'],2)
my_database_comp[,'P100']=round(my_database_comp[,'P100'],2)
```
```{r, echo=FALSE}
my_database_comp1 <- my_database_comp [,c(1,2,3,5,6,7,8,9)]
```
```{r, echo=FALSE}
knitr::kable(head(my_database_comp1, n = 5), format = "html", caption = "Aminoacid composition") %>% 
  kable_styling("striped", full_width = F) %>%
  column_spec(6:8, bold=T, color= "black") %>% 
  footnote(general= "totalAA = Number of total amino acids of the protein
           Qcomp= Number of Glutamine found in the protein 
           Q100= Porcentage of Glutamine in the protein
           Pcomp= Number of Proline found in the protein 
           P100= Porcentage of Proline in the protein",
           footnote_as_chunk = T)
```

**Images from a file**

![](resources/img/tidyverse.png){width=100px}

**Plots from R**
```{r standard-plot, out.width='60%', fig.align='center', fig.height= 4, fig.width=6, fig.cap="Yet another gapminder plot"}
gapminder %>% 
  filter(year == 1987) %>% 
  ggplot(aes(x = gdpPercap, y = lifeExp, colour = continent, size = pop)) +
  geom_point() +
  scale_x_log10(labels = function(b) format(b, scientific = F)) +
  scale_size(trans = "sqrt", guide = "none") +
  scale_color_brewer(palette = "Set1") +
  theme_linedraw() +
  theme(legend.position = "bottom")
```

Your figure and table captions are automatically numbered and can be referenced in the text
if needed: see eg. Table \@ref(tab:mytable) and Figure \@ref(fig:standard-plot)

# My Digital Toolbox

What digital tools have you been using in your project? Do you expect that everything will be able 
to be completed within R, or will you need to work with multiple tools to get the right result?
Which of the digital skills needed for your project have you learned since starting Data School?

You can use all the usual R markdown features in writing a project summary, including lists:

* R - dplyr, ggplot, ...
* Python
* SQL

## Favourite tool (optional)

Is there a tool/package/function in particular that you've enjoyed using? Give it a special shout out here. What about this tool makes it your favourite?

![](https://raw.githubusercontent.com/tidyverse/ggplot2/master/man/figures/logo.png){.pull-right width=100px}

No prizes for guessing mine:

# My time went ...

What parts of your project take the most time and effort? Were there any surprising challenges you
encountered, and how did you solve them?

# Next steps

What further steps do you wish your project could take? Or are there any new digital skills that you
are keen to develop as a result of your involvement in the Data School?

# My Data School Experience

This summary is mostly about your project. However we would also like to hear about other
parts of your Data School experience. What aspects of the program did you really enjoy? Have you
tried applying the skills you have learned in your daily work? Have you been able to transfer this 
knowledge to your team members? Any descriptions of the personal impact the program has 
had are welcome here as well!
